web-ng/src/app/components/map/map.component.ts

```diff
@@ -124,21 +120,26 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         );
       case EditMode.AddPolygon:
         // TODO: Implement adding polygon.
+        if (!selectedLayerId) {
+          return Promise.resolve();
+        }
         return Promise.resolve();
       case EditMode.None:
       default:
+        this.navigationService.setFeatureId(null);
```

Comment:
Nit: Can we make `setFeatureId()` not accept nulls and add a method `clearFeatureId()`? This would make the code more self-documenting (by not using `null` as special value in the internal API)

Reply:
Can I then expose `clearFeatureId()` as a public function in `NavigationService`, and then inside `NavigationService`, `clearFeatureId()` is calling `this.setFeatureId(null)`?

Reply:
Yes, you could do it that way provided the public API doesn't accept `null` (ie public `selectFeature()` and `clearFeature` both call `private setFeature(id)`. Though if you're overriding the entire fragment, I suspect the logic will be simple enough that a common method (`setFeature`) will not be needed.

Reply:
SG. Done.